Source/src/importers/PrefabImporter.cpp

```diff
@@ -109,10 +91,60 @@ Hachiko::UID Hachiko::PrefabImporter::CreatePrefabAsset(const char* name, GameOb
 
     root->CollectObjectsAndComponents(object_collector, component_collector);
     root->SavePrefabReferences(prefab->prefab_data, object_collector, component_collector);
+}
+
+void Hachiko::PrefabImporter::LoadScriptReferences(ResourcePrefab* prefab, GameObject* root)
+{
+    // Wire script references by indexes
+    std::vector<const GameObject*> object_collector = std::vector<const GameObject*>();
+    std::vector<const Component*> component_collector = std::vector<const Component*>();
+    // add a nullptr to 0 to reserve index 0 for missing references
+    object_collector.push_back(nullptr);
+    component_collector.push_back(nullptr);
+
+    root->CollectObjectsAndComponents(object_collector, component_collector);
+    // For load references we dont need to pass the yaml data since it is already parsed
+    // And assigned to each script on their normal load
+    root->LoadPrefabReferences(object_collector, component_collector);
+
+    for (auto script : root->GetComponents(Component::Type::SCRIPT))
+    {
+        static_cast<Scripting::Script*>(script)->OnLoad();
+    }
+    for (auto script : root->GetComponentsInDescendants(Component::Type::SCRIPT))
+    {
+        static_cast<Scripting::Script*>(script)->OnLoad();
+    }
+}
+
+Hachiko::UID Hachiko::PrefabImporter::CreatePrefabAsset(const char* name, GameObject* root)
+{
+    // This uid doesnt mater since it is a temporary resource
+    ResourcePrefab* prefab = CreatePrefabResouce(name, 0, root);
     
     const std::string prefab_asset_path = StringUtils::Concat(GetResourcesPreferences()->GetAssetsPath(Resource::AssetType::PREFAB), prefab->name, PREFAB_EXTENSION);
     FileSystem::Save(prefab_asset_path.c_str(), prefab->prefab_data);
-    UID prefab_uid = App->resources->ImportAssetFromAnyPath(prefab_asset_path)[0];
     delete prefab;
+    UID prefab_uid = App->resources->ImportAssetFromAnyPath(prefab_asset_path)[0];
+    
     return prefab_uid;
-}
\ No newline at end of file
+}
+
+
+Hachiko::UID Hachiko::PrefabImporter::CreateGeneratedPrefab(const char* name, UID uid, GameObject* root)
+{
+    // Pre creates the prefab meta so it already contains the desired uid decided by model importer
+    ResourcePrefab* prefab = CreatePrefabResouce(name, uid, root);
+    const std::string prefab_asset_path = std::string(ASSETS_FOLDER_GENERATED_PREFAB) + std::to_string(uid) + PREFAB_EXTENSION;
```

Comment:
consider using StringUtils::Concat

Reply:
Is it for readability or performance reasons? (Thx for ur reviews btw <3)